src/integrationTest/resources/envelope.json

```diff
@@ -4,18 +4,18 @@
       "case_number": "1111222233334446",
       "po_box": "SSCSPO",
       "jurisdiction": "BULKSCAN",
-      "delivery_date": "2018-06-23T00:00:00.000Z",
-      "opening_date": "2018-06-23T00:00:00.000Z",
+      "delivery_date": "2018-06-23T02:10:30.123Z",
+      "opening_date": "2018-06-23T02:10:30.123Z",
       "zip_file_name": "7_24-06-2018-00-00-00.zip",
       "classification": "exception",
       "scannable_items": [
         {
           "document_control_number": "1111002",
-          "scanning_date": "2018-06-23T00:00:00.000Z",
+          "scanning_date": "2018-06-23T02:10:30.123Z",
           "ocr_accuracy": "test",
           "manual_intervention": "test",
           "next_action": "return",
-          "next_action_date": "2018-06-23T00:00:00.000Z",
+          "next_action_date": "2018-06-24T00:00:00Z",
```

Comment:
Is this the agreed format?

Reply:
let keep the format as it was, this story was supposed to be about the class used internally.

Reply:
`2018-06-23T00:00:00.000Z` the time values are all zeros here. `Instant` class ignores the milliseconds for such values

Reply:
This is not agreed format `2018-06-23T02:10:30.123Z` should be agreed one. I think json schema validation will fail ?

Reply:
@nitinprabhu I see the same format in all test files and the validation is not failing

Reply:
If you check the json schema it is ```"pattern": "^\\d{4}\\-\\d{2}\\-\\d{2}T\\d{2}\\:\\d{2}\\:\\d{2}\\.\\d{3}Z$"``` so 3 digits for seconds not sure why the validation doesn't fails ?

Reply:
Looks like we will need to add more tests because I am sure when we did end to end tests that format won't work

Reply:
This is not metadata fail. this is the result of parsed envelope, which will be put on the service bus queue

Reply:
it works even if the values are 0. So `2018-06-23T00:00:00.000Z` works

Reply:
yes it works. please see the metadata files used for the test.
src/integrationTest/resources/zipcontents/mismatching_pdfs/metadata.json
src/integrationTest/resources/zipcontents/ok/metadata.json

Reply:
I just uploaded a zip file with date format as  "2018-12-27T10:46:03.12Z",but it has thrown schema validation error like below.
"Failed validation for file 65755555551369_06-12-2018-07-22-36.zip against schema. Errors:
 error: ECMA 262 regex "^\d{4}\-\d{2}\-\d{2}T\d{2}\:\d{2}\:\d{2}\.\d{3}Z$" does not match input string "2018-12-27T10:46:03.12Z" level: "error" schema: {"loadingURI":"#","pointer":"/properties/delivery_date"} instance: {"pointer":"/delivery_date"} domain: "validation" keyword: "pattern" regex: "^\\d{4}\\-\\d{2}\\-\\d{2}T\\d{2}\\:\\d{2}\\:\\d{2}\\.\\d{3}Z$" string: "2018-12-27T10:46:03.12Z" error: ECMA 262 regex "^\d{4}\-\d{2}\-\d{2}T\d{2}\:\d{2}\:\d{2}\.\d{3}Z$" does not match input string "2018-12-27T10:46:03.12Z" level: "error" schema: "

Reply:
if it works with what it was, let's not change it

Reply:
`2018-12-27T10:46:03.12Z` does not follow any of the formats discussed above, @vakepati

Reply:
> does `2018-06-23T00:00:00.000Z` fail?
Passed!